File: 卡码/代码随想录/Python/贪心/03-困难/LeetCode_0053.py
#53. 最大子数组和
from typing import *

class Solution:
  def maxSubArray(self, nums: List[int]) -> int:
    res:int=float("-inf")
    cnt:int=0
    for num in nums:
      cnt+=num
      if cnt>res:
        res=cnt
      if cnt<0:
        cnt=0
    return res

File: 卡码/代码随想录/JS/贪心/03-困难/LeetCode_0053.js
//53. 最大子数组和
const maxSubArray = function(nums) {
  let res=-Infinity,cnt=0;
  for(let num of nums){
    cnt+=num;
    if(cnt>res) res=cnt;
    if(cnt<0) cnt=0;
  }
  return res;
};